apps/sms/views/shared/js/activity_handler.js

```diff
@@ -88,51 +90,51 @@ var ActivityHandler = {
           return attachment;
         });
 
-        var size = attachments.reduce(function(size, attachment) {
-          if (attachment.type !== 'img') {
-            size += attachment.size;
-          }
-
-          return size;
-        }, 0);
-
-        if (size > Settings.mmsSizeLimitation) {
-          Utils.alert({
-            id: 'attached-files-too-large',
-            args: {
-              n: activityData.blobs.length,
-              mmsSize: (Settings.mmsSizeLimitation / 1024).toFixed(0)
-            }
-          }).then(() => ActivityClient.postResult());
-          return;
-        }
-
-        dataToShare = attachments;
+        data = attachments;
         break;
       case ActivityDataType.URL:
-        dataToShare = activityData.url;
+        data = activityData.url;
         break;
       // As for the moment we only allow to share one vcard we treat this case
       // in an specific block
       case ActivityDataType.VCARD:
-        dataToShare = new Attachment(activityData.blobs[0], {
+        data = new Attachment(activityData.blobs[0], {
           name: activityData.filenames[0],
           isDraft: true
         });
         break;
       default:
-        ActivityClient.postError(
+        return ActivityClient.postError(
           'Unsupported activity data type: ' + activityData.type
         );
-        return;
     }
 
-    if (!dataToShare) {
-      ActivityClient.postError('No data to share found!');
-      return;
+    if (!data) {
+      return ActivityClient.postError('No data to share found!');
     }
 
-    return this.toView({ body: dataToShare });
+    // Treat the non array data (url, vcard) size as 0
```

Comment:
Nope as far as I know. I'll verify this again and file a bug if vcard size is still missing.